### PR TITLE
Added split op bf16/fp32 oneDNN kernel

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added split op bf16/fp32 oneDNN kernel. This PR caused 20% speed up on DPN68 model, measured on Intel(R) Core(TM) i9-9940X CPU @ 3.30GHz with 1000 repeated iterations. For bigger speed up keys optimizations are needed. For now OneDNN kernel will be used only if AxisTensor and SectionsTensorList are not set.

### CODE REVIEW

File: paddle/fluid/operators/split_op.cc
Comment (jakpiase) [inline paddle/fluid/operators/split_op.cc:80]: removed this function just for testing purposes, it needs to be restored and changed before this PR will be merged

**Review by jczaja — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/mkldnn/split_mkldnn_op.cc
Comment (lidanqing-intel) [inline paddle/fluid/operators/mkldnn/split_mkldnn_op.cc:55]: Hi  I just wonder why code block could not be compute directly?

File: paddle/fluid/operators/split_op.cc
Comment (lidanqing-vv) [inline paddle/fluid/operators/split_op.cc:150]: There is no need to add following?
```
AddAttr<std::string>(
      "mkldnn_data_type",
      "(string, default \"float32\"). Data type of mkldnn kernel")
      .SetDefault("float32")
      .InEnum({"float32", "int8", "bfloat16"});
```
then cpu_bfloat16_placement_pass.cc will set it "bfloat16" and then in bf16 model inference, bf16 kernel will called? 
The 20% is FP32 perf improved right?

Comment (jczaja) [reply paddle/fluid/operators/split_op.cc:150]: I can only comment that reported 20% is  oneDNN execuition without split mkldnn vs with split mkldnn.  Performance of bf16 was not evaluated according to my knwoledge.

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/split_op.cc
Comment (wozna) [reply paddle/fluid/operators/split_op.cc:150]: Yes please add this attribute because it is needed for BF16 inference, but you can add enum without "int8": 
``` 
AddAttr<std::string>(
      "mkldnn_data_type",
      "(string, default \"float32\"). Data type of mkldnn kernel")
      .SetDefault("float32")
      .InEnum({"float32", "bfloat16"});
```

File: python/paddle/fluid/tests/unittests/mkldnn/test_split_bf16_mkldnn_op.py
Comment (wozna) [inline python/paddle/fluid/tests/unittests/mkldnn/test_split_bf16_mkldnn_op.py:30]: Please set the `self.mkldnn_data_type` attribute to` bfloat16` so that `check_dygraph` will be False by default. 
This way, you won't need a separate approval for this.

**Review by wozna — COMMENTED** [review]
It would be good if you adapted these changes to the functionality of inference BF16.
This involves:
- adding `split` to the support list in the file: https://github.com/PaddlePaddle/Paddle/blob/773aabc7710f030b7b2e83d30637cd94c3379194/paddle/fluid/framework/ir/graph_pattern_detector.cc#L2264
- adding the `mkldnn_data_type` attribute for this operator.

File: python/paddle/fluid/tests/unittests/mkldnn/test_split_bf16_mkldnn_op.py
Comment (jakpiase) [reply python/paddle/fluid/tests/unittests/mkldnn/test_split_bf16_mkldnn_op.py:30]: Thank you, now I understand the purpose of "mkldnn_data_type" attribute, I'll add that

**Review by jczaja — APPROVED** [review]

**Review by wozna — APPROVED** [review]
LGTM

**Review by lidanqing-vv — APPROVED** [review]
LGTM.

**Review by ghost — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Pull request merged**
